doc/src/guides/solving/solve-matrix-equation.md

```diff
@@ -0,0 +1,102 @@
+# Solve a Matrix Equation
+
+Use SymPy to solve a matrix (linear) equation. For example, solving $
+\left[\begin{array}{cc} c & d\\1 & -e\end{array}\right] \left[\begin{array}{cc}
+x\\y\end{array}\right] = \left[\begin{array}{cc} 2\\0\end{array}\right] $ yields
+$ \left[\begin{array}{cc} x\\y\end{array}\right] = \left[\begin{array}{cc}
+\frac{2e}{ce+d}\\\frac{2}{ce+d}\end{array}\right]$.
+
+Alternatives to consider:
+- Solving a matrix equation is equivalent to solving a system of linear
+  equations, so if you prefer you can use *link to solve a system of equations
+  once published*
+- *alternative 2*
+
+Here is an example of solving a matrix equation with SymPy's {func}`~.LUsolve`.
+We use the standard matrix equation formulation $Ax=b$ where
+- $A$ is the matrix representing the coefficients in the linear equations
+- $x$ is the column vector of unknowns to be solved for
+- $b$ is the column vector of constants, where each row is the value of an
+  equation
+
+```py
+>>> from sympy import symbols
+>>> from sympy.matrices import Matrix
+>>> c, d, e = symbols("c, d, e")
+>>> A = Matrix([[c,d], [1, -e]])
+>>> A
+Matrix([
+[c,  d],
+[1, -e]])
+>>> b = Matrix([2, 0])
+>>> b
+Matrix([
+[2],
+[0]])
+>>> A.LUsolve(b)
```

Comment:
Should we recommend using [`LUsolve`](https://docs.sympy.org/dev/modules/matrices/matrices.html#sympy.matrices.matrices.MatrixBase.LUsolve) or [`solve`](sympy.matrices.matrices.MatrixBase.solve)? Both are from `sympy.matrices.matrices.MatrixBase.LUsolve` and give the same result, at least in this example.

Reply:
Referring to the source code, [`solve`](https://github.com/sympy/sympy/blob/master/sympy/matrices/solvers.py#L702) calls methods such as `LUsolve` and `gauss_jordan_solve`. Because `solve` is the parent function, I'll go with it unless anyone replies otherwise.

Reply:
@oscarbenjamin, request your input on any situations where we should recommend a different method (than the default Gauss-Jordan elimination) be used to solve a matrix equation (aka set of linear equations).

Reply:
The advantage of an LU decomposition is that you can solve a system with the same matrix repeatedly for different left hand sides more efficiently than doing a full solve each time.

Another possibility is that you might want to compute the inverse matrix and then use that to generate solutions.

Reply:
Thanks. Are [these good mathematical approaches for LUdecomposition and inverting](https://github.com/sympy/sympy/pull/23951/files#diff-df986b323ac7cf904576dea15ea8555e9fcf73ed504d1d459e86a5078251743bR68)?

Reply:
Added checks that solutions are correct by multiplying: A * solution = b.